tests/kernel/io_uring.pm

```diff
@@ -0,0 +1,141 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes liburing testing suite
+# Maintainer: Andrea Cervesato <andrea.cervesato@suse.com>
+# More documentation is at the bottom
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+
+sub run {
+    my $self = shift;
+
+    select_serial_terminal;
+
+    my $repository = get_var('LIBURING_REPO', 'https://github.com/axboe/liburing.git');
+    my $timeout = get_var('LIBURING_TIMEOUT', 1800);
+    my $version = get_var('LIBURING_VERSION', '');
+    my $exclude = get_var('LIBURING_EXCLUDE', '');
+    my $issues = get_var('LIBURING_KNOWNISSUES', '');
+    my @lines;
+
+    # select latest liburing version which is supported by the system
+    if ($version eq '') {
+        my $pkginfo = script_output("zypper --no-refresh info liburing2");
```

Comment:
This is a bit tricky indeed. We need to test as much functionalities as possible with liburing2 testing suite, but we don't know what kernel io_uring patches or features we have. For this reason, code always takes the latest library available version, which is supposed to test **enough** functionalities.

Reply:
Yes, `rpm -qf liburing2` is what we want. Also, I'd use put info into `record_info()` so that it stays even logs are removed from openQA.

Reply:
How it will handle following case?
```
Information for package liburing2:
----------------------------------
Repository     : Main Repository (OSS)
Name           : liburing2
Version        : 2.3-2.2
Arch           : x86_64
Vendor         : openSUSE
Installed Size : 66.6 KiB
Installed      : Yes (automatically)
Status         : out-of-date (version 2.3-2.1 installed)
Source package : liburing-2.3-2.2.src
```
Do you see it?

```
rpm -q liburing2
liburing2-2.3-2.1.x86_64
```